.github/workflows/pr-lint.yml

```diff
@@ -39,6 +39,14 @@ jobs:
       - name: Verify Golang Modules
         run: make BUILD_IN_CONTAINER=false generate-modules-ci
 
+      - name: Run make generate
+        run: make generate
+
+      # to ensure we don't miss any files that were supposed to be autogenerated.
+      # if there's a diff then the workflow will exit here.
+      - name: Run verify manifests
+        run: make verify-manifests
+
```

Comment:
the other calls to `make` use BUILD_IN_CONTAINER. Why do the two new calls to make not have it?

Why not call `make verify` and shorten the github actions file (for example remove the Lint Docker, Lint YAML)?

Reply:
I'm in for using `make verify` too. This will make things easier for us. 

In case, we decide to use `make verify` we can also consider dropping pr-lint.yaml workflow because then `lint` will be embedded inside verify as well. 

See the following target: 
```make
ALL_VERIFY_CHECKS = boilerplate shellcheck starlark manifests
.PHONY: verify
verify: generate lint $(addprefix verify-,$(ALL_VERIFY_CHECKS)) ## Verify all
```

Reply:
Please check https://github.com/syself/cluster-api-provider-hetzner/commit/e3a412591b52238d4cf24b3fdef220cdd691a55d 

Via this commit, I removed individual ones and used `make verify` directly and some other ones directly suited for CI.